src/Native/System.Native/pal_errno.cpp

```diff
@@ -339,20 +347,19 @@ extern "C" int32_t SystemNative_ConvertErrorPalToPlatform(Error error)
         case PAL_EXDEV:
             return EXDEV;
         case PAL_ENONSTANDARD:
-            break; // fall through to assert
+        default:
+            // We should not use this function to round-trip platform -> pal
+            // -> platform. It's here only to synthesize a platform number
+            // from the fixed set above. Note that the assert is outside the
+            // switch rather than in a default case block because not
+            // having a default will trigger a warning (as error) if there's
+            // an enum value we haven't handled. Should that trigger, make
+            // note that there is probably a corresponding missing case in the
+            // other direction above, but the compiler can't warn in that case
+            // because the platform values are not part of an enum.
+            assert(false && "Unknown error code");
```

Comment:
@jasonwilliams200OK, we wanted this assert and return outside of the switch.  The point of doing that was exactly because we wanted the compiler to issue a warning if we neglected to include an appropriate case statement.  Please revert this.